test/setup/cluster_setup.go

```diff
@@ -279,6 +280,56 @@ func (c *Cluster) StartServer(s *Server) error {
 	return s.start(c.cmdTimeout)
 }
 
+func (c *Cluster) GetServerByID(serverID string) (*Server, int) {
+	if serverID == "" {
+		return nil, -1
+	}
+	for i, srv := range c.Servers {
+		if srv.serverID == serverID {
+			return srv, i
+		}
+	}
+	return nil, -1
+}
+
+func (c *Cluster) AgreedLeader(t *testing.T, activeServers ...int) int {
+	var leaders [3]int
+	var leader int
+
+	for _, srvVal := range activeServers {
+		clusterStatusResEnv, err := c.Servers[srvVal].QueryClusterStatus(t)
+		if err == nil && clusterStatusResEnv != nil {
+			_, leader = c.GetServerByID(clusterStatusResEnv.GetResponse().GetLeader())
+			if leader == -1 {
+				return -1
+			}
+			leaders[srvVal] = leader
```

Comment:
you can simply start with an empty `leaders`  slice and append to it every time you find a leader, than, at the end, just check the slice has the same leader. i.e. there would be no gaps in that slice even if the `activeServers` does contain gaps.

Reply:
done